File: scripts/crosscheck.ts
// TS 포팅본이 Python 원본과 같은 판정을 내는지 대조한다.
// 사용: npx tsx scripts/crosscheck.ts docs/evidence/photo
import fs from "node:fs";
import path from "node:path";
import { verify } from "../lib/verify";

const dir = process.argv[2] ?? "docs/evidence/photo";
for (const f of fs.readdirSync(dir).sort()) {
  if (!f.endsWith(".json")) continue;
  const r = verify(JSON.parse(fs.readFileSync(path.join(dir, f), "utf8")));
  const name = f.replace("agent-raw-", "").replace(".json", "");
  console.log(`${name}\t${r.verdict}\t${r.checksPassed ?? "-"}/${r.checksTotal ?? "-"}`);
}

File: scripts/crosscheck.py
#!/usr/bin/env python3
"""Python 원본 판정을 TS 대조용 같은 형식으로 출력한다."""
import importlib.util, json, os, sys
spec = importlib.util.spec_from_file_location("v", "scripts/verify.py")
v = importlib.util.module_from_spec(spec); spec.loader.exec_module(v)
d = sys.argv[1] if len(sys.argv) > 1 else "docs/evidence/photo"
for f in sorted(os.listdir(d)):
    if not f.endswith(".json"): continue
    r = v.verify(json.load(open(os.path.join(d, f))))
    name = f.replace("agent-raw-", "").replace(".json", "")
    print(f"{name}\t{r['verdict']}\t{r.get('checks_passed','-')}/{r.get('checks_total','-')}")
